src/modules/url-preview/generatePreview.js

```diff
@@ -7,16 +7,18 @@ const strategies = [
 ];
 
 export default async function generatePreview(url, cb) {
-	return new Promise((resolve) => {
+	return new Promise(async (resolve) => {
```

Comment:
You don't need to wrap an async function in a promise constructor. `return` instead of `resolve`